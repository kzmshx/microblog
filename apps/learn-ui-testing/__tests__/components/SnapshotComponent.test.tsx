/**
 * - スナップショットテストの仕組み
 *   - 初回にスナップショットを作成し、次回以降はスナップショットと比較して差分があればテストが失敗する。
 *   - これにより意図しないDOMの変更を検知できる。
 *   - 意図したDOMの変更をしたい場合は `vitest -u` でスナップショットを更新する。
 */

import { render } from "@testing-library/react";
import SnapshotComponent from "../../app/components/SnapshotComponent";

describe("SnapshotComponent", () => {
  it("should pass snapshot test", () => {
    const { container } = render(<SnapshotComponent text="Hello, snapshot test!" />);
    expect(container).toMatchSnapshot();

    // const { container } = render(<SnapshotComponent text="Yeah, snapshot text!" />);
    // expect(container).toMatchSnapshot();
  });
});
