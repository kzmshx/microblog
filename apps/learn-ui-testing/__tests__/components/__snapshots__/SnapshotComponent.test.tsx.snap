// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SnapshotComponent > should pass snapshot test 1`] = `
<div>
  <div>
    <p>
      Hello, snapshot test!
    </p>
  </div>
</div>
`;
